File: framework/java/src/main/java/soda/web/exception/ServiceException.java
package soda.web;

public class ServiceException extends RuntimeException {

    private int bizCode;

    private String errMessage;

    private int httpCode;

    public ServiceException(int bizCode, String errMessage, int httpCode, Throwable cause) {
        super(errMessage, cause);
        this.bizCode = bizCode;
        this.errMessage = errMessage;
        this.httpCode = httpCode;
    }

    public ServiceException(int bizCode, String errMessage, int httpCode) {
        super(errMessage);
        this.bizCode = bizCode;
        this.errMessage = errMessage;
        this.httpCode = httpCode;
    }

    public ServiceException(int bizCode, String errMessage) {
        this(bizCode, errMessage, 500);
    }

    public int getBizCode() {
        return bizCode;
    }

    public String getErrorMessage() {
        return errMessage;
    }

    public int getHttpCode() {
        return httpCode;
    }

}

File: framework/kotlin/src/main/kotlin/soda/kotlin/web/exception/ServiceException.kt
package soda.kotlin.web.exception

open class ServiceException(
    val bizCode: Int,
    val errMessage: String,
    val httpCode: Int,
    cause: Throwable?
) : RuntimeException(errMessage, cause) {

    constructor(bizCode: Int, errMessage: String, httpCode: Int):
            this(bizCode, errMessage, httpCode, null) {}

    constructor(bizCode: Int, errMessage: String):
            this(bizCode, errMessage, 500) {}

}